### ISSUE TITLE
Feature request: Show hg revision tag in the About dialog

### ISSUE THREAD

**Issue by pychess** [pychess/pychess#420]
Original [issue 420](https://code.google.com/p/pychess/issues/detail?id=420) reported by [igor2x](https://code.google.com/u/116691631290437288902/) 2009-03-19

It would be nice if Help | About would display number of version from
subversion. I have multiple development versions and also stable one (to
compare what has been fixed) and it would be nice to know which version is
version started at the moment. It would also make me easy to know which is
installed version.

**Comment by pychess** [comment]
Original [comment 1.](https://code.google.com/p/pychess/issues/detail?id=420#c1) posted by [igor2x](https://code.google.com/u/116691631290437288902/) 2009-03-20

Note: can you please change title of this "issue" to: "feature requests: add version
in Help | About"

**Comment by pychess** [comment]
Original [comment 2.](https://code.google.com/p/pychess/issues/detail?id=420#c2) posted by [gbtami](https://code.google.com/u/gbtami/) 2009-03-20
- changed summary to: feature requests: svn version in Help | About
- added label: Enhancement

**Comment by pychess** [comment]
Original [comment 3.](https://code.google.com/p/pychess/issues/detail?id=420#c3) posted by [lobais](https://code.google.com/u/lobais/) 2009-03-20

We can probably read the revision number from the hidden subversion files you get
when you do an svn checkout or update.

I have found ./svn/entries in the first lines:
8

dir
1419
http://pychess.googlecode.com/svn/trunk
http://pychess.googlecode.com/svn

I don't know if this would always work. E.g. I don't know if subversion updates the
file when you do a commit.
Have anybody got more subversion knowledge? Can we just use line 4 from this file?

**Comment by pychess** [comment]
Original [comment 4.](https://code.google.com/p/pychess/issues/detail?id=420#c4) posted by [lobais](https://code.google.com/u/lobais/) 2009-03-20
- changed status to: Accepted

**Comment by pychess** [comment]
Original [comment 6.](https://code.google.com/p/pychess/issues/detail?id=420#c6) posted by [igor2x](https://code.google.com/u/116691631290437288902/) 2009-03-21

Revision number would be nice.
Yesterday it was 1419 revision number, now gbtami has updated my translation and I
have downloaded svn version and now it is 1420.

**Comment by pychess** [comment]
Original [comment 7.](https://code.google.com/p/pychess/issues/detail?id=420#c7) posted by [lobais](https://code.google.com/u/lobais/) 2009-03-21

Ok, I've given it a shot (r1421)
It seams to work at least with the current version of subversion.

For developers, it doesn't automatic update when you commit, but after next update
(even though nothing is new), it will.
- changed status to: Fixed

**Comment by pychess** [comment]
Original [comment 8.](https://code.google.com/p/pychess/issues/detail?id=420#c8) posted by [igor2x](https://code.google.com/u/116691631290437288902/) 2009-03-21

Hi,
if game is run from pychess-read-only using command "PYTHONPATH=lib/ ./pychess" then
this revision number is correctly displayed as: "PyChess Staunton r1425".

But if game is installed from svn files using command: "python setup.py install"
and then installed PyChess is run the incorrectly label is displayed as: "PyChess
Staunton 0.10alpha2".

It would be nice when game get installed to preserve revision number. I also think it
would be nice to have also version number label, example:
"PyChess Staunton 0.10alpha2 r1425".

This will tell more: it is 0.10 version (not 0.8) and it is version after alpha2.

**Comment by pychess** [comment]
Original [comment 9.](https://code.google.com/p/pychess/issues/detail?id=420#c9) posted by [lobais](https://code.google.com/u/lobais/) 2009-03-22

Actually I've never really used 'python setup.py install'. Won't it overwrite old
installations?
- changed status to: Reopened

**Comment by pychess** [comment]
Original [comment 10.](https://code.google.com/p/pychess/issues/detail?id=420#c10) posted by [igor2x](https://code.google.com/u/116691631290437288902/) 2009-03-22

That is right. It will overwrite old installation. But if everything is working file
with svn version, then I overwrite old installation, because I don't need two working
versions.

**Comment by pychess** [comment]
Original [comment 11.](https://code.google.com/p/pychess/issues/detail?id=420#c11) posted by [lobais](https://code.google.com/u/lobais/) 2009-03-22

I see.

For transfering the revision number to an installation, we could either create some
install script, which copies the svn file or creates something similar;
Or we could make PyChess write to the const.py file when run.

None of them are very nice though.
The install script solution is probably the least ugly, and it doesn't require you to
run PyChess before the sync happens.

**Comment by pychess** [comment]
Original [comment 12.](https://code.google.com/p/pychess/issues/detail?id=420#c12) posted by [igor2x](https://code.google.com/u/116691631290437288902/) 2009-03-23

I think the install script should take over this issue.

**Comment by pychess** [comment]
Original [comment 13.](https://code.google.com/p/pychess/issues/detail?id=420#c13) posted by [igor2x](https://code.google.com/u/116691631290437288902/) 2010-09-04

I have checked this feature request and I think this is not needed anymore. The latest revision info can be accesses using: "hg tip" command. I needed revision number before update. If something goes wrong with new code I can always go back. So this problem is solved by "hg tip" command.

You can close this bug report. Not needed anymore.

**Comment by pychess** [comment]
Original [comment 14.](https://code.google.com/p/pychess/issues/detail?id=420#c14) posted by [lobais](https://code.google.com/u/lobais/) 2010-09-04

Well, it would still be nice to have.
But of course it should now be hg revision name, rather than svn :)
- changed summary to: Feature request: Show hg revision tag in the About dialog
- added label: Low

**Comment by pychess** [comment]
Original [comment 15.](https://code.google.com/p/pychess/issues/detail?id=420#c15) posted by [cqbkaju](https://code.google.com/u/109969498147979196868/) 2010-10-20

I see your point.  If your running the current hg pychess, Help > About still says "Staunton 0.10beta4" but "hg tip" says:
changeset:   1663:05a03c8c0177
tag:         tip
user:        gbtami
date:        Tue Oct 19 11:16:36 2010 +0200
summary:     Use deferred translation for ping error message

Maybe it should say something like "Staunton 0.10beta4 rev. 1663" ?

**Comment by pychess** [comment]
Original [comment 16.](https://code.google.com/p/pychess/issues/detail?id=420#c16) posted by [mattgatto](https://code.google.com/u/mattgatto/) 2010-10-21

This one would be easy to fix for anyone that's interested ;). The code that needs to be changed is at pychess/lib/pychess/Main.py:

```
    if os.path.isfile(prefix.addDataPrefix(".svn/entries")):
        f = open(prefix.addDataPrefix(".svn/entries"))
        line4 = [f.next() for i in xrange(4)][-1].strip()
        widgets["aboutdialog1"].set\_version(VERSION\_NAME+" r"+line4)
    else:
        widgets["aboutdialog1"].set\_version(VERSION\_NAME+" "+VERSION)
```

P.S. to Thomas: could you add an "easy-fix" Label to the Issue Tracker?

**Comment by pychess** [comment]
Original [comment 17.](https://code.google.com/p/pychess/issues/detail?id=420#c17) posted by [cqbkaju](https://code.google.com/u/109969498147979196868/) 2010-10-21

Easy to fix?  If people are willing to wait, I would love to give it a try this weekend.

Lobais kindly gave me some pointers on how to help with building dialogs for board settings.
I hope to start that soon as well.

**Comment by pychess** [comment]
Original [comment 18.](https://code.google.com/p/pychess/issues/detail?id=420#c18) posted by [igor2x](https://code.google.com/u/116691631290437288902/) 2010-10-21

cqbkaju: "Maybe it should say something like "Staunton 0.10beta4 rev. 1663"?"

No, this is not correct! Local revision ID should never be used in Mercurial to explicitly define revision number. Mercurial is distributed source control system where local revision ID can vary between cloned repositories - so each user can have different local revision ID. Don't confuse this revision ID with central source control systems like Subversion where there is exactly ONE revision ID.

Because local revision ID in Mercurial can be different between computers, the only really reliable number is to use global revision ID which is hexadecimal number.

Commands:
- get local revision ID: hg id -n
- get global revision ID: hg id -i
- help about ID: hg help id

I suggest to use "global revision ID" to explicitly define revision number.

**Comment by pychess** [comment]
Original [comment 19.](https://code.google.com/p/pychess/issues/detail?id=420#c19) posted by [igor2x](https://code.google.com/u/116691631290437288902/) 2010-10-21

@command 18:

Sample:
"hg id -n" outputs: 1664
"hg id -i" outputs: e584be69f120 --> this number should be used to unique identify revision

**Comment by pychess** [comment]
Original [comment 20.](https://code.google.com/p/pychess/issues/detail?id=420#c20) posted by [cqbkaju](https://code.google.com/u/109969498147979196868/) 2010-10-22

@ igor2x

Thank you for the clarification.  As you have surmised, I am not very familiar with Mercurial.

But from my point of view, using the hex code "hg id -i" gives us presents a new problem.
That number is basically meaningless to most humans.

Perhaps a more usable solution would be something like:

"Staunton 0.10beta4"
   rev. e584be69f120
   2010-OCT-22"

In other words, we add the ID as well as the date.
Then everyone knows roughly when the code is pulled?

Or even easier (?) leave the ID out and just keep the date.
That way we can leave the date in for regular releases without changing the code?

"Staunton 0.10beta4"
   2010-OCT-22"

Thoughts?

Thanks,
Dave

**Comment by pychess** [comment]
Original [comment 21.](https://code.google.com/p/pychess/issues/detail?id=420#c21) posted by [igor2x](https://code.google.com/u/116691631290437288902/) 2010-10-22

You are right that hex number is not very friendly, but it uniquely identifies revision.

It would be nice to have some kind of date, but pull date is something that is also not unique. In Mercurial pull happens when end-user "downloads" source code from central Google repository to local repository. Because each user can pull (download) code in different time, this time would probably be incorrect.

The main purpose of revision number is to identify which version end-user is using when submitting a bug report. So we could put in bug report text something like: "Please write down the version number that you can access from Help | About menu." This would give a fast info to developers and other contributors to take some testing to find out if reported bug is really a bug or false report. And if we look into this info in this kind of way then hex number really means something, and it means something to developers/contributors.

**Comment by pychess** [comment]
Original [comment 22.](https://code.google.com/p/pychess/issues/detail?id=420#c22) posted by [mattgatto](https://code.google.com/u/mattgatto/) 2011-04-02
- added label: Easy-Fix

**Comment by pychess** [comment]
Original [comment 23.](https://code.google.com/p/pychess/issues/detail?id=420#c23) posted by [gbtami](https://code.google.com/u/gbtami/) 2012-11-08
- changed status to: Fixed
